### PR TITLE
Issues/introduce holidays

### PR DESCRIPTION
**Pull request by AndreyDegtyarev**
Introduced Holiday entity.
Refactored code.

### CODE REVIEW

File: src/Nox.Reference.Data.World/Migrations/WorldDbContextModelSnapshot.cs [hunk 1]
```diff
@@ -331,50 +354,39 @@ protected override void BuildModel(ModelBuilder modelBuilder)
                     b.ToTable("NumberFormat");
                 });
 
-            modelBuilder.Entity("Nox.Reference.Data.World.Entities.TimeZones.TimeZone", b =>
```
Comment (ILoveMyTortoise) [inline src/Nox.Reference.Data.World/Migrations/WorldDbContextModelSnapshot.cs:334]: where is time zone code?

File: src/Nox.Reference.Data.World/DataContext/WorldInfo.cs [hunk 2]
```diff
@@ -38,9 +38,12 @@ public static IQueryable<IVatNumberDefinitionInfo> VatNumberDefinitions
...
 
-    private static IWorldInfoContext DbContext
-        => new WorldDbContext(new DbContextOptions<WorldDbContext>(), _mapper, _configuration);
+    public static IQueryable<ICountryHolidayInfo> Holidays
+        => DbContext.Holidays;
 
     public static IQueryable<ICultureInfo> Cultures
         => DbContext.Cultures;
+
+    private static IWorldInfoContext DbContext
```
Comment (ILoveMyTortoise) [inline src/Nox.Reference.Data.World/DataContext/WorldInfo.cs:47]: should we always create new? Maybe have one static?

File: src/Nox.Reference.Data.World/DataContext/WorldDbContext.cs [hunk 3]
```diff
@@ -36,32 +39,33 @@ public WorldDbContext()
...
     public IQueryable<ICurrencyInfo> Currencies
-        => Set<Currency>()
-             .AsNoTracking()
+        => GetData<Currency, CurrencyInfo>();
+
+    public IQueryable<ICultureInfo> Cultures
+        => Set<Culture>()
              .AsQueryable()
-             .ProjectTo<CurrencyInfo>(_mapper.ConfigurationProvider);
+             .ProjectTo<CultureInfo>(_mapper.ConfigurationProvider);
```
Comment (ILoveMyTortoise) [inline src/Nox.Reference.Data.World/DataContext/WorldDbContext.cs:47]: why not getData here?

**Review by ILoveMyTortoise — APPROVED** [review]

File: src/Nox.Reference.Data.World/DataContext/WorldDbContext.cs (hunk 3 above)
Comment (AndreyDegtyarev) [reply src/Nox.Reference.Data.World/DataContext/WorldDbContext.cs:47]: Fixed in other  pr.

File: src/Nox.Reference.Data.World/Mappings/HolidayMapping.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+            //In
+            CreateMap<CountryHolidayInfo, CountryHoliday>().ReverseMap();
+            CreateMap<HolidayDataInfo, HolidayData>().ReverseMap();
+            CreateMap<StateHolidayInfo, StateHoliday>().ReverseMap();
+            CreateMap<LocalHolidayNameInfo, LocalHolidayName>().ReverseMap();
+            CreateMap<RegionHolidayInfo, RegionHoliday>().ReverseMap();
+
+            //Out
+            CreateProjection<CountryHoliday, CountryHolidayInfo>();
+            //CreateProjection<HolidayData, HolidayDataInfo>();
```
[the remark this answers is not in the document]
Comment (AndreyDegtyarev) [reply src/Nox.Reference.Data.World/Mappings/HolidayMapping.cs]: Done.

**Pull request merged**